src/NHibernate/Type/GenericOrderedSetType.cs

```diff
@@ -24,9 +27,23 @@ public GenericOrderedSetType(string role, string propertyRef)
 		{
 		}
 
+		static ConstructorInfo GetSetConstructor()
+		{
+			var type = System.Type.GetType("Iesi.Collections.Generic.LinkedHashSet`1, Iesi.Collections", false);
+			if (type == null)
+				throw new TypeLoadException(
+					@"Could not load Iesi.Collections.Generic.LinkedHashSet`1
+
+Please make sure that you have referenced the Iesi.Collections package.");
```

Comment:
I'm planning to add a release note about this implicit dependency in a "Possible breaking change" part. I'll try to reword the exception to provide more clarity.